### DISCUSSION TITLE
Fundamentos do ASP.NET 6 - Erro interno no servidor

### DISCUSSION THREAD

**Discussion by Samuel-Lucas** [balta-io/2811#11] [Q&A] [answered]
Olá, estou na parte testar as requisições do CRUD de categorias, não estou conseguindo ter resposta dos métodos GET, POST, PUT e DELETE quando executo o servidor no terminal e faço a requisição no Postman, aparece esse erro:

`System.InvalidOperationException: The constraint reference ' int' could not be resolved to a type. Register the constraint type with 'Microsoft.AspNetCore.Routing.RouteOptions.ConstraintMap'.`

Fiz umas pesquisas, será que tenho que fazer uma configuração em Program.cs de mapeamento das constraints tal como diz nesse link ?
[https://social.msdn.microsoft.com/Forums/en-US/a5365899-271b-4387-8149-1a6f9b80fee7/invalidoperationexception-the-constraint-reference-customer-could-not-be-resolved-to-a-type?forum=aspdotnetcore](url)

Precisa criar um arquivo de configure services ou tem uma outra maneira para fazer em Program.cs ?

Repositório da minha aplicação: [https://github.com/Samuel-Lucas/Blog_ASP6/tree/master](url)

Arquivo Program.cs
![program](https://user-images.githubusercontent.com/49723833/153759056-7c997df3-7112-462b-b27a-1b2850e5ce68.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Samuel-Lucas, como vai?

Não há necessidade de fazer configuração no Program.cs neste momento. Talvez esteja faltando algum código no seu controller.
Chegou a comparar seu código com o daqui do GitHub? Acho que passou algo...

Qualquer coisa, posta o código do seu controller aqui que ajudamos.

**Reply by Samuel-Lucas** [reply]
Olá tudo bem, parece que o problema foi porque os parâmetros com a tipagem para a rota do método precisam estar juntos:

![image](https://user-images.githubusercontent.com/49723833/154173206-aada2a7d-2f23-4334-9f2d-31a5014fdd44.png)

Antes estava `[HttpGet("v1/categories/{id: int}")]` com um espaço depois dos ':'

Achei estranho, estava pensando que era algo a ser corrigido com instalação de pacote ou configuração, mas foi por formato :/

O link onde estava procurando e achei a correção: [https://stackoverflow.com/questions/62667806/getting-system-invalidoperationexception-the-constraint-reference-int-could](url)

**Reply by andrebaltieri** [reply]
Que bom que deu certo então @Samuel-Lucas !
